test/strings/basic.jl

```diff
@@ -529,3 +530,113 @@ end
 # unrecognized escapes in string/char literals
 @test_throws ParseError parse("\"\\.\"")
 @test_throws ParseError parse("\'\\.\'")
+
+# prevind and nextind tests
+
+let str = "∀α>β:α+1>β"
+    @test prevind(str, -1) == -2
+    @test prevind(str, -1, 1) == -2
+    @test prevind(str, 1) == 0
+    @test prevind(str, 1, 1) == 0
+    @test prevind(str, 2) == 1
+    @test prevind(str, 2, 1) == 1
+    @test prevind(str, 4) == 1
+    @test prevind(str, 4, 1) == 1
+    @test prevind(str, 5) == 4
+    @test prevind(str, 5, 1) == 4
+    @test prevind(str, 5, 2) == 1
+    @test prevind(str, 5, 3) == 0
+    @test prevind(str, 15) == 14
+    @test prevind(str, 15, 1) == 14
+    @test prevind(str, 15, 2) == 13
+    @test prevind(str, 15, 3) == 12
+    @test prevind(str, 15, 4) == 10
+    @test prevind(str, 15, 10) == 0
+    @test prevind(str, 15, 9) == 1
+    @test prevind(str, 15, 10) == 0
+    @test prevind(str, 16) == 15
+    @test prevind(str, 16, 1) == 15
+    @test prevind(str, 16, 2) == 14
+    @test prevind(str, 20) == 15
+    @test prevind(str, 20, 1) == 15
+    @test prevind(str, 20, 10) == 1
+    @test_throws ArgumentError prevind(str, 20, 0)
+
+    @test nextind(str, -1) == 1
+    @test nextind(str, -1, 1) == 1
+    @test nextind(str, 0, 2) == 4
+    @test nextind(str, 0, 20) == 17
+    @test nextind(str, 0, 10) == 15
+    @test nextind(str, 1) == 4
+    @test nextind(str, 1, 1) == 4
+    @test nextind(str, 1, 2) == 6
+    @test nextind(str, 1, 9) == 15
+    @test nextind(str, 1, 10) == 17
+    @test nextind(str, 2) == 4
+    @test nextind(str, 2, 1) == 4
+    @test nextind(str, 3) == 4
+    @test nextind(str, 3, 1) == 4
+    @test nextind(str, 4) == 6
+    @test nextind(str, 4, 1) == 6
+    @test nextind(str, 14) == 15
+    @test nextind(str, 14, 1) == 15
+    @test nextind(str, 15) == 17
+    @test nextind(str, 15, 1) == 17
+    @test nextind(str, 20) == 21
+    @test nextind(str, 20, 1) == 21
+    @test_throws ArgumentError nextind(str, 20, 0)
+end
+
+let str = GenericString("∀α>β:α+1>β")
```

Comment:
Couldn't this block be merged with the previous one using a `for` loop?

Reply:
Unfortunately no - `nextind` and `prevind` return something different in corner cases for `String` and `AbstractString`. Both are correct wrt the contract given in the docstring but give different results.

Reply:
But maybe have a common part and a separate part, so that it's easier to spot the differences?